### DISCUSSION TITLE
tailwindcss  plugins.forEach is not a function

### DISCUSSION THREAD

**Discussion by hijack-621** [tailwindlabs/tailwindcss#8191] [Help]
I referred  the Installation for [Tailwind CLI](https://tailwindcss.com/docs/installation)【called  method 1】 and [Using PostCSS](https://tailwindcss.com/docs/installation/using-postcss) 【called method 2】.

method 1 may success installing, but it genetator a  css file that only have over 500 rows code in it. use like `bg-red-xxx text-xl` is useless
method 2 may occurred errors, it show like `UnhandledPromiseRejectionWarning: TypeError: plugins.forEach is not a function `.

so how to successfully use tailwindcss in html with link tag?   ths a lot here

**Comment by johnryan** [comment]
In my case this was caused by misreading the directions and putting:

```
plugins: {
    tailwindcss: {},
    autoprefixer: {},
  }
```
inside tailwind.config.js 

instead of `plugins: [],` as the instructions say

**Reply by cosmicwanderer7** [reply]
Try plugins: ["tailwindcss ,autoprefixer"],

**Comment by Anjade** [comment]
Doesnt help. The error is still there.

**Comment by cosmicwanderer7** [comment]
Try 
`plugins: ["tailwindcss ,autoprefixer"], `
instead of 
```
plugins: {
    tailwindcss: {},
    autoprefixer: {},
  }
```

**Comment by Karhide** [comment]
If you go to the [postcss-loader docs](https://github.com/webpack-contrib/postcss-loader) you'll see how they do it nowadays:

**postcss.config.js**
```
module.exports = {
  plugins: [
    [
      "postcss-preset-env",
      {
        // Options
      },
    ],
  ],
};
```

This worked for me.

**Comment by hijack-621** [comment]
tks，man。



---- Replied Message ----
| From | John ***@***.***> |
| Date | 05/04/2022 05:47 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [tailwindlabs/tailwindcss] tailwindcss plugins.forEach is not a function (Discussion #8191) |

In my case this was caused by misreading the directions and putting:

plugins: {
    tailwindcss: {},
    autoprefixer: {},
  }


inside tailwind.config.js

instead of plugins: [], as the instructions say

**Comment by NiyongaboEric** [comment]
Or try this way:

```/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./src/**/*.{html, js}"],
  theme: {
    extend: {},
  },
  plugins: [
    {
      tailwindcss: {},
      autoprefixer: {},
    },
  ],
};
```

**Comment by orbit2c** [comment]
This is the way:  

plugins: [
    require('tailwindcss'),
    require('autoprefixer'),
  ]
